tests/cluster/etl/test_transformer.py

```diff
@@ -25,6 +25,10 @@ def test_process_many(self):
         self.assertEqual({"TestTransformer"}, set(result.keys()))
         self.assertEqual(6, list(result.values())[0].count())
 
+    def test_dataset_key(self):
+        result = TestTransformer("my_key").etl({"df": self.df})
```

Comment:
create these test in a new testfile which only tests "ExtendedTransformer".

Reply:
This test is just an addition to testing the functionality of the core transformer class. I noticed that there really was no test of the dataset_key functionality so i added it with this PR. Want me to remove?

Reply:
No - please let it stay. I thought it had something to do with the extended version. Good job!